### PR TITLE
[DS-4257] (One Webapp Phase 2) Rename "dspace-spring-rest" to "dspace-server-webapp"

### PR DESCRIPTION
**Pull request by tdonohue**
https://jira.duraspace.org/browse/DS-4257

This is **Phase 2** of the [DSpace Backend as One Webapp](https://wiki.duraspace.org/display/DSPACE/DSpace+Backend+as+One+Webapp) code refactor.  It simply does the following:

* Rename `[src]/dspace-spring-rest` module/directory to `[src]/dspace-server-webapp`
* Rename `[src]/dspace/modules/spring-rest` overlay module/directory to `[src]/dspace/modules/server`
* Update POM description of `dspace-rest` module to make it clear it is the **deprecated** REST API
* Change all other mentions of "spring-rest" into "server"
    * NOTE: In a few instances I added reference to a new `dspace.server.url` configuration that does NOT yet exist. That will come in Phase 3 (see below).

The *end result* is that this DSpace backend webapp now deploys to http://localhost:8080/server/ by default.

I've tried to keep this PR to ONLY renaming this webapp/module.  This helps make it easy to review (as most of the changes here are file renames), and also attempts to avoid larger scale code conflicts with other PRs.

After this Phase 2 PR is merged, there *will be a Phase 3 PR*.  

In **Phase 3**, I'll do the following:
1. Create the new `dspace.server.url` configuration in `dspace.cfg`
2. Replace _all usages_ of `dspace.url`, `dspace.baseUrl` and `dspace.restURL` (in the codebase & tests) with `dspace.server.url`

### CODE REVIEW

**Review by AlexanderS — COMMENTED** [review]
The renaming looks good. I don't like the reference of a non-existent config variables, but it should be okay if phase 3 follows in near future.

**Review by mwoodiupui — APPROVED** [review]
Builds without error.  All tests succeed.  (Strange:  see comments from AlexanderS regarding missing 'server/' in two ITs.)  Works with Angular FE.  PMH responder works.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/RelationshipRestRepositoryIT.java [hunk 1]
```diff
@@ -237,8 +237,8 @@ public void createRelationshipWriteAccessLeftItem() throws Exception {
                                                                (org.springframework.data.rest.webmvc.RestMediaTypes
                                                                     .TEXT_URI_LIST_VALUE))
                                                            .content(
-                                                               "https://localhost:8080/spring-rest/api/core/items/" + publication.getID() + "\n" +
-                                                                   "https://localhost:8080/spring-rest/api/core/items/" + author1.getID()))
+                                                               "https://localhost:8080/api/core/items/" + publication.getID() + "\n" +
```
[the remark this answers is not in the document]
Comment (tdonohue) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/RelationshipRestRepositoryIT.java]: This has now been resolved.  It turns out this URL is only being used in a unit-style test, which is why the test will succeed even with the missing `server/` path.  Still, I added it for consistency.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/UriListParsingIT.java [hunk 2]
```diff
@@ -71,8 +71,8 @@ public void mockRequestTextUriParsingTest() throws Exception {
 
 
         MockHttpServletRequest mockRequest = new MockHttpServletRequest();
-        String uriListString =  "https://localhost:8080/spring-rest/api/core/items/" + publicItem1.getID() + "\n" +
-            "https://localhost:8080/spring-rest/api/core/items/" + publicItem2.getID();
+        String uriListString =  "https://localhost:8080/api/core/items/" + publicItem1.getID() + "\n" +
+            "https://localhost:8080/api/core/items/" + publicItem2.getID();
```
[the remark this answers is not in the document]
Comment (tdonohue) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/UriListParsingIT.java]: This has now been resolved.  It turns out this URL is only being used in a unit-style test, which is why the test will succeed even with the missing `server/` path.  Still, I added it for consistency.

**Comment by tdonohue** [comment]
This has now been rebased & @AlexanderS's comments were resolved.  Ready again for final review & testing.

**Comment by terrywbrady** [comment]
After this is merged, we will need to modify the following line.

- https://github.com/DSpace-Labs/DSpace-Docker-Images/blob/master/add-ons/angular-tools/environment.dev.js#L7

**Review by terrywbrady — APPROVED** [review]
Tested with Docker.  The changes look good.

**Review by AlexanderS — APPROVED** [review]
Thanks for fixing the paths. Everything looks fine, now.

**Comment by tdonohue** [comment]
As agreed in today's DSpace 7 meeting, this is ready to merge.  Merging

**Pull request merged**